bin/aws_write_creds.sh

```diff
@@ -25,16 +21,21 @@ if [ ${#missing_auth_vars[@]} -ne 0 ]
 then
     echo "Did not find values for:" 
     printf ' %q\n' "${missing_vars[@]}"
-    echo "Will assume they are in credentials file"
+    echo "Will assume they are in credentials file or not needed"
 else
     echo "Creating credentials file"
     # Create the directory....
     mkdir -p ~/.aws
     CREDS=~/.aws/credentials
     echo "[default]"                                    >  $CREDS
     echo "aws_access_key_id=$AWS_ACCESS_KEY_ID"         >> $CREDS 
-    echo "aws_secret_access_key=$AWS_SECRET_ACCESS_KEY" >> $CREDS 
-    echo "aws_session_token=$AWS_SESSION_TOKEN"         >> $CREDS 
+    echo "aws_secret_access_key=$AWS_SECRET_ACCESS_KEY" >> $CREDS
+    # This is if we have non-temp credentials...
+    if [[ -z "${AWS_SESSION_TOKEN+x}" ]]; then
+      echo "Not using AWS Session Token"
```

Comment:
optional nit: Would it be more diagnosable to say "AWS_SESSION_TOKEN was empty or unset. Not using AWS Session Token".  That way there's a clear distinction between "the script has decided not to use an aws session token and "we checked this var and it was empty so we're continuing without it".  Might help in the case where the user thought they'd set it but didn't.

Reply:
Yeah, that sounds quite a bit better than what I had...let me make a change so it is more clear.